### ISSUE TITLE
Assignment w15 -- RBL final presentation

### ISSUE THREAD

**Issue by dudung** [dudung/sk5004-01-2024-2#7]
## Final Project Presentation Instructions
You will present the progress of your project in a 5-minute presentation. Please ensure your presentation includes the following:

+ Team Information
  - Names of all team members
+ Project Overview
  - Source of your data
  - Method or approach you plan to use
  - Machine learning category (e.g., supervised learning, unsupervised learning, etc.)
  - Estimated timeline for completing the project
+ Content
    Include the key elements of your project so far:
  - Problem definition – What problem are you trying to solve?
  - Proposed solution – Briefly explain your approach or idea
  - Discussion – Early insights, challenges, or decisions made so far
  - Understanding – Demonstrate that your team understands the topic, data, and method
+ Task Assignment
Specify which team member is responsible for each of the following outputs (each item should be assigned to at least one member):    
  - Medium story
  - YouTube video
  - Gist code
  - OSF presentation slides

## After the Presentation
Even if your team does not get a chance to present, please submit the following details:

### Submission Requirements:
+ Student ID and full name of all team members
+ Project details as listed above (team members, data source, method, ML category, estimated timeline, and task assignments)

### Submission Format:
Submit all the requested information as a comment in the assigned GitHub issue.

No additional links or attachments are needed.

**Comment by giovaldiramadhan** [comment]
Nama : Giovaldi Ramadhan
NIM : 20924303
Project: Photometric Redshift Estimation using Regression Models


## Project Details: 

1. Team Members:
a. Muhammad Rizaldi Yani H (20924301)
b. Giovaldi Ramadhan (20924303)
c. Aldiansyah Anugrah Ramadhan (20924304)
d. Gia Muhammad Agusta (20924305)

2. Data Source: SDSS Data Release 18
3. Method:

```mermaid
graph TD;
    A[Start] --> B[Query Data from Database]
    B --> C[Data Preprocessing]
    C --> D[75% Train]
    C --> E[25% Test]

    D --> F1[Extra Tree Regressor]
    D --> F2[Random Forest]
    D --> F3[Deep Learning]

    F1 --> G1["Evaluate (R2, MAE, MSE)"]
    F2 --> G2["Evaluate (R2, MAE, MSE)"]
    F3 --> G3["Evaluate (R2, MAE, MSE)"]

    G1 --> H[Final Model]
    G2 --> H
    G3 --> H

    H --> I["Hyperparameter Tuning"]
    I --> J["Evaluate (R2, MAE, MSE)"]
    J --> K[Test Model]
    E --> K

    K --> L[Output Model]
    L --> M[End]
```

Notes
* Data Preprocessing (filtering zWarning = 0, excluding the 'Star' class, handling outliers, filtering color index, train test split)
* Deep Learning (Deep Convolutional Neural Network)
* Hyperparameter Tuning (Random Search and Genetic Algorithm)
* Output Model (Redshift)

4. ML Category: Supervised Learning (Regression)
5. Estimated Timeline After the first RBL Presentation:

| Activities                | 28 Mei | 29 Mei | 30 Mei | 31 Mei | 1 Jun | 2 Jun | 3 Jun | 4 Jun | 5 Jun | 6 Jun | 7 Jun | 8 Jun | 9 Jun |
|---------------------------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|
| Hyperparameter Tuning Model | 🟩     |  🟩    | 🟩     | 🟩     | 🟩     | 🟩     | 🟩     |       |        |        |        |        |        |
| Output Making             |      |        |        |        |        | 🟧     | 🟧     | 🟧     | 🟧     | 🟧     | 🟧     | 🟧     |        |
| Publish Output            |        |      |        |        |        |        |        |        |        |        |        |        | 🟪     |


6. Task Assignments:
a. Medium Story: Rizaldi and Gia
b. YouTube video: All members
c. Git code: Aldiansyah
d. OSF Presentation slides: Giovaldi

**Comment by Dikifernandi18** [comment]
Name : DIKI FERNANDI
NIM : 20124023
PROJECT TITLE : PREDICTING NANOFIBER AVERAGE DIAMETER USING MACHINE LEARNING

OUR TEAMS:
1.       Diki Fernandi 20124023
2.       Aulia Defitri Wulandari 20224001
3.       Kamilah Nada Maisa 20224010
4.       Sudarminto Senlau 20224007

for complete Final Project details please access : https://github.com/Dikifernandi18/RBL-MACHINE-LEARNING/blob/main/Team%20Information.pdf

**Comment by uliviaembun** [comment]
Name : Ulivia Embun Tresna Wardani
NIM : 10322015
Project Title: Machine Learning Approach to NGC 7790 Membership Determination

## Project Detail
* **Team members**:
    * Muhammad Fathir Rizqy Fadli (10322005)
    * Ulivia Embun Tresna Wardani (10322015)
    * Naura Kasihna Risat (10322037)
    * Awang Shandy Candradewani (10322042)

* **Data Source**:  database survey GAIA DR3 via TOPCAT

* **Methods & ML Category**:
NGC 7790 data that was obtained via TOPCAT was processed using four different machine learning algorithms in order to compare the differences in result / performance between them and then the best would be chosen. The following are ML algorithms that being used and it's technical justification.
    1. Random Forest (Supervised)
        Random Forest has emerged as a particularly effective algorithm for stellar membership determination due to its ensemble learning approach (using many decision trees to make final decision) thus it resistant to overfitting. It particularly suitable for astronomical datasets where training samples may be limited or contain inherent uncertainties.
    2. Support Vector Machine (Supervised)
        SVM is selected due to its superior performance in high-dimensional spaces, ability to create non-linear decision boundaries through kernel transformations, robustness to outliers through the use of support vectors makes it well-suited for astronomical data. In astronomical data, the parameter space often includes multiple astrometric and photometric dimensions where cluster and field star populations may not be linearly separable. It also often contains measurement errors and contamination from field stars. The algorithm's ability to find optimal separating hyperplanes even in the presence of noise is particularly valuable when dealing with faint cluster members at the detection limits.
    3. K-means (Unsupervised)
        Both previous methods require arguably a lot of computation power. Sometimes astronomers wouldn't mind getting lower accuracy / performance if they could save a lot of time in exchange, that's why we bring in K-means algorithms. K-means clustering has been extensively adopted in astronomical applications due to its computational efficiency and straightforward implementation. The algorithm's ability to handle large datasets makes it particularly suitable for modern surveys like Gaia, which contain billions of stellar measurements. Beside that, centroid-based approach of K-means aligns well with the physical reality of stellar clusters, where member stars tend to cluster around common values in parameter spaces such as proper motion and parallax.
    4. HDBScan (Unsupervised)
        HDBSCAN represents a significant advancement over traditional density-based clustering methods, making it particularly well-suited for stellar cluster analysis. Its ability to identify clusters of varying densities addresses a fundamental challenge in stellar astronomy, where cluster cores may be much denser than their outer regions. The hierarchical approach of HDBSCAN also enable it to naturally captures the multi-scale structure of stellar clusters, from tight central concentrations to extended halos. This capability is crucial because stellar clusters often exhibit complex spatial structures that cannot be adequately described by single-density thresholds.4.⁠ ⁠⁠ML category: Supervised and unsupervised learning.

* **Estimated timeline after presentation:** 
```mermaid
%%{
  init: {
    'theme': 'light',
    'themeVariables': {
      'primaryColor': '#2E86C1',
      'ganttCrit': '#9B59B6',
      'ganttDone': '#90EE90',
      'ganttActive': '#48C9B0',
      'taskTextDarkColor': '#000000',
      'taskTextLightColor': '#FFFFFF',
      'lineColor': '#D5D8DC'
    }
  }
}%%
gantt
    title Timeline Output Making - RBL Machine Learning and AI
    dateFormat  YYYY-MM-DD
    axisFormat %b %Y

    section Final Presentation
    D-day:done, 2025-06-04, 1d
    
    section Project Finalization
    Revision for better output (5-9 June)     :active, 2025-06-05, 5d

    section Output Making
    Medium Story (7-23 June)     :active, 2025-06-07, 17d
    YouTube video (9-23 June)     :active, 2025-06-09, 15d
    Git code (9-23 June)    :active, 2025-06-09, 15d
    OSF Presentation slides (1-23 June):  :active, 2025-06-01, 23d

    section Publishing Output
    D-day (24 June):crit,2025-06-24, 2d
```

* **Task Assignment**
    * Medium Story: Awang assigned
    * YouTube video: Ulivia and Naura assigned
    * Git code: Fathir assigned
    * OSF Presentation slides: Ulivia & Naura assigned

**Comment by yulianaAzizah** [comment]
- Team Member:
   - Yuliana Nur Azizah (20224004)
   - Husna Farida (20224006)
   - Adilah Luthfi (20224024)

- Data Source : SBA Loan Dataset

- Method

```mermaid
flowchart TD

    A[Start]
    B[Dataset]
    C[Data Preprocessing]
    D[Data Transformation]
    E1[Train 80%]
    E2[Test 20%]
    F[Apply SMOTE]

    M1[Logistic Regression]
    M2[K-Nearest Neighbors]
    M3[Decision Tree]
    M4[Random Forest]
    M5[XGBoost]

    G[Model Predict]
    H[Model Evaluation]
    I[Final Model]
    J[Prediction Result]
    K[End]

    A --> B
    B --> C
    C --> D
    D --> E1
    D --> E2
    E1 --> F 
    F --> M1
    F --> M2
    F --> M3
    F --> M4
    F --> M5

    M1 --> G
    M2 --> G
    M3 --> G
    M4 --> G
    M5 --> G

    G --> H
    H --> I
    E2 --> I
    I --> J
    J --> K
```

- ML Category : Supervised Learning (Classification)

### Estimated Timeline After the First RBL Presentation

| Agenda               | June w1 | w2  | w3  |      w4      |
|----------------------|---------|-----|-----|--------------|
| Final Presentation   | ✅      |     |     |              |
| Project Finalization |         | ✅  | ✅  | ✅           |
| Publish              |         |     |     | ✅ (22)      |

- Task Assignment  
  Medium Story: Husna Farida  
  YouTube video: All Member  
  Git code: Yuliana Nur Azizah  
  OSF Presentation slides: Adilah Luthfi

**Comment by Habiburr0hman** [comment]
## Team Members
1. Melvan Savero Lee (10121063)
2. Habiburrohman (10121089)

## Project Overview
### Source
PlantVillage Dataset
### Method
Convolutional Neural Network (CNN)
### Machine Learning Category
Supervised Learning
### Timeline
| **Steps**               | **March** | **April** | **May** | **June, 1-16** |
|---------------------------|-----------|-----------|---------|----------|
| **Literature Review**       |     v     |     v     |         |          |
| **Data Gathering**      |           |     v     |         |          |
| **Data Preprocessing**  |           |     v     |         |          |
| **Model Building**  |           |     v     |    v    |          |
| **Model Optimization (Hyperparameter Tuning)** |           |           |    v    |     v    |
| **Result Arranging and Documentation**      |           |           |         |     v    |

## Content
### Problem Definition
Tomato was an essential nutrition source for human. However, tomato cultivation is very susceptible to various diseases caused by microorganism infections. The symptom of disease infection usually present in the leaf for the first time. Conventional tomato cultivation monitoring and disease detection activities are highly dependent on visual observation. This procedure can take a log time, especially if the agricultural land is very large. 
### Proposed Solution
We propose tomato leaf disease classification using deep learning method called Convolutional Neural Network (CNN). The resulting model can help the farmers to automate monitoring activities of tomato cultivation so the diseased plant can be detected and treated as soon as possible. 
### Discussion
From the data, there are different disease that has similar symptom. This is the challenge for us to create a CNN model that can distinguish the disease correctly. 
### Understanding
The dataset we use is subset of PlantVillage dataset. From other various plant, we only take the tomatoes becasue the the number of disease is largest among others. We use CNN because of its ability to extract hidden feature of the image data through convolutional and pooling layers. 

## Task Assignment
Melvan: Medium Article, OSF Presentation Slide, YouTube Video
Habib: GitHub Repository, Medium Article, OSF Presentation Slide, Youtube Video

**Comment by naiilard** [comment]
### Team Members
1. Ari Ratna Gumilang (20123311)
2. Elizabet Sri Rezeki (20123016)
3. Hari Julian Putra (20123008)
4. Naila Ratu Dianti (20123019)

### Method
Principal Component Analysis

### Machine Learning Category
Unsupervised Learning

### Data Source
PRIMS Indonesia and FIRMS NASA

### Timeline
| 16 April | 17-30 April | 1-27 Mei | 28 Mei | 29 Mei-3 Juni | 4 Juni | 5-17 Juni |
|--------|--------|--------|--------|--------|--------|--------|
| Presentasi awal | Pengolahan data awal | Pembuatan ppt | Presentasi hasil | Revisi | Presentasi akhir | Upload tugas |

### Task Assignment
Link Medium : https://medium.com/@nailaratudianti/key-factors-behind-indonesias-peatland-fires-3219e8e7c14b (by: Naila Ratu Dianti)
Link GitHub : https://github.com/ElizabetSR/Tugas-Machine-Learning/blob/main/Principal_Component_Analysis.ipynb (by: Elizabet Sri Rezeki)
Link YouTube : https://youtu.be/NznxnGeEbKc?si=ju8XWiQgDPwkaXaU (by: Hari Julian Putra)
Link OSF : https://osf.io/tzhdy (by: Ari Ratna Gumilang)
